tests/components/tasmota/test_device_trigger.py

```diff
@@ -0,0 +1,832 @@
+            "subtype": "switch_1",
+        },
+    ]
+    expected_triggers2 = copy.deepcopy(expected_triggers1)
+    expected_triggers2[1]["type"] = "button_double_press"
+
+    triggers = await async_get_device_automations(hass, "trigger", device_entry.id)
+    for expected in expected_triggers1:
+        assert expected in triggers
+
+    # Update trigger
+    async_fire_mqtt_message(hass, f"{DEFAULT_PREFIX}/{mac}/config", json.dumps(config2))
+    await hass.async_block_till_done()
+
+    triggers = await async_get_device_automations(hass, "trigger", device_entry.id)
+    for expected in expected_triggers2:
+        assert expected in triggers
+
+    # Remove trigger
+    async_fire_mqtt_message(hass, f"{DEFAULT_PREFIX}/{mac}/config", json.dumps(config3))
+    await hass.async_block_till_done()
+
+    triggers = await async_get_device_automations(hass, "trigger", device_entry.id)
+    assert triggers == []
+
+
+async def test_if_fires_on_mqtt_message(
+    hass, device_reg, calls, mqtt_mock, setup_tasmota
+):
+    """Test triggers firing."""
+    # Discover a device with 2 device triggers
+    config = copy.deepcopy(DEFAULT_CONFIG)
+    config["swc"][0] = 0
+    config["swc"][2] = 9
+    mac = config["mac"]
+
+    async_fire_mqtt_message(hass, f"{DEFAULT_PREFIX}/{mac}/config", json.dumps(config))
+    await hass.async_block_till_done()
+    device_entry = device_reg.async_get_device(set(), {("mac", mac)})
+
+    assert await async_setup_component(
+        hass,
+        automation.DOMAIN,
+        {
+            automation.DOMAIN: [
+                {
+                    "trigger": {
+                        "platform": "device",
+                        "domain": DOMAIN,
+                        "device_id": device_entry.id,
+                        "discovery_id": "00000049A3BC_switch_1_TOGGLE",
+                        "type": "button_short_press",
+                        "subtype": "switch_1",
+                    },
+                    "action": {
+                        "service": "test.automation",
+                        "data_template": {"some": ("short_press")},
+                    },
+                },
+                {
+                    "trigger": {
+                        "platform": "device",
+                        "domain": DOMAIN,
+                        "device_id": device_entry.id,
+                        "discovery_id": "00000049A3BC_switch_3_HOLD",
+                        "subtype": "switch_3",
+                        "type": "button_double_press",
+                    },
+                    "action": {
+                        "service": "test.automation",
+                        "data_template": {"some": ("long_press")},
+                    },
+                },
+            ]
+        },
+    )
+
+    # Fake short press.
+    async_fire_mqtt_message(hass, "tasmota_49A3BC/stat/SWITCH1T", '{"TRIG":"TOGGLE"}')
+    await hass.async_block_till_done()
+    assert len(calls) == 1
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6333224)